### PR TITLE
[kf5windowsystem] fix cmake.in for static usage; extra fixes

### PR DESCRIPTION
**Pull request by wrobelda**
- #### What does your PR fix?  
    - Apply https://invent.kde.org/frameworks/kwindowsystem/-/merge_requests/27 patch
    - fix MAYBE_UNUSED_VARIABLES usage
    - use semVer
    - update deprecated functions
    - wrap paths in quotes
    - don't overzealously remove folders and actually remove those that need to be

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?  
  No change.

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?  
  Yes.

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?  
  Yes.

### CODE REVIEW

File: ports/kf5windowsystem/27.patch [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+@@ -4,7 +4,13 @@ include(CMakeFindDependencyMacro)
+ find_dependency(Qt5Gui @REQUIRED_QT_VERSION@)
+ 
+ if(NOT @KWINDOWSYSTEM_NO_WIDGETS@)
+-find_dependency(Qt5Widgets @REQUIRED_QT_VERSION@)
++    find_dependency(Qt5Widgets @REQUIRED_QT_VERSION@)
++endif()
++
++if(NOT @BUILD_SHARED_LIBS@)
++    if(@WIN32@)
```
Comment (JackBoosY) [inline ports/kf5windowsystem/27.patch:23]:
```suggestion
+    if(WIN32)
```

**Review by JackBoosY — CHANGES_REQUESTED** [review]

File: ports/kf5windowsystem/27.patch (hunk 1 above)
Comment (wrobelda) [reply ports/kf5windowsystem/27.patch:23]: Why this suggestion? This was already merged upstream and confirmed working.

File: ports/kf5windowsystem/27.patch [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+diff --git a/KF5WindowSystemConfig.cmake.in b/KF5WindowSystemConfig.cmake.in
+index 6c1b4c0..bcb7128 100644
+--- a/KF5WindowSystemConfig.cmake.in
++++ b/KF5WindowSystemConfig.cmake.in
+@@ -4,7 +4,13 @@ include(CMakeFindDependencyMacro)
+ find_dependency(Qt5Gui @REQUIRED_QT_VERSION@)
+ 
+ if(NOT @KWINDOWSYSTEM_NO_WIDGETS@)
+-find_dependency(Qt5Widgets @REQUIRED_QT_VERSION@)
++    find_dependency(Qt5Widgets @REQUIRED_QT_VERSION@)
```
Comment (Neumann-A) [inline ports/kf5windowsystem/27.patch:19]: Tell upstream this is not how one should find Qt. 
The correct syntax is to use
 `find_(packacke|dependency)(Qt[MajorVersion] <Version> COMPONENTS Widgets)`
Never should you directly lookup cmake configs of components.

Comment (wrobelda) [reply ports/kf5windowsystem/27.patch:19]: Looks like the opinions vary. Mind providing some reference?
Screen Shot 2021-08-09 at 10 09 50 PM

Comment (dg0yt) [reply ports/kf5windowsystem/27.patch:19]: Repeated calls to `find_package(<PKG> ...  COMPONENT ...)` don't or didn't work as expected is some cases. For example, `<PKG>_FOUND` is set to true after the first call, so further calls are ignored. This is an issue when components (sub-directories) in a project do their own dependency lookup.

Comment (dg0yt) [reply ports/kf5windowsystem/27.patch:19]: And for completeness, until Qt 5.11, `find_package(Qt5Widgets)` etc. was the recommended form:
https://doc.qt.io/archives/qt-5.11/cmake-manual.html

File: ports/kf5windowsystem/27.patch (hunk 1 above)
Comment (wrobelda) [reply ports/kf5windowsystem/27.patch:23]: @JackBoosY

File: ports/kf5windowsystem/27.patch (hunk 2 above)
Comment (JackBoosY) [reply ports/kf5windowsystem/27.patch:19]: Was this resolved?

Comment (wrobelda) [reply ports/kf5windowsystem/27.patch:19]: @JackBoosY this is a discussion around the upstream code quality, unrelated to this port.

**Review by JackBoosY — APPROVED** [review]

**Pull request merged**